### PR TITLE
New modbus data types

### PR DESCRIPTION
**Pull request by janiversen**
## Proposed change

Correct intending error.
Make common configuration for sensor/climate
Add description of new data types

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
https://github.com/home-assistant/core/pull/52423
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/modbus.markdown [hunk 1]
```diff
@@ -56,6 +56,16 @@ name:
   required: false
   default: "modbus_hub"
   type: string
+retries:
+  description: Number of times to retry a request.
+  required: false
+  default: 3
+  type: int
```
Comment (klaasnicolaas) [inline source/_integrations/modbus.markdown]:
```suggestion
  type: integer
```

File: source/_integrations/modbus.markdown [hunk 2]
```diff
@@ -220,11 +230,60 @@ slave:
...
+structure:
+      description: "If `data_type` is custom specified a double-quoted Python struct is expected here, to format the string to unpack the value. See Python documentation for details. Example: `>i`."
+      required: false
+      type: string
+      default: ">f"
+swap:
+      description: "Swap the order of bytes/words, options are `none`, `byte`, `word`, `word_byte`."
+      required: false
+      default: none
+      type: string
```
Comment (klaasnicolaas) [inline source/_integrations/modbus.markdown]: I see that you use here 4/5 indents but it should be 2 😉

**Review by klaasnicolaas — DISMISSED** [review]

File: source/_integrations/modbus.markdown (hunk 1 above)
Comment (janiversen) [reply source/_integrations/modbus.markdown]: Done

File: source/_integrations/modbus.markdown (hunk 2 above)
Comment (janiversen) [reply source/_integrations/modbus.markdown]: done

**Review by nikito7 — APPROVED** [review]

**Review by frenck — APPROVED** [review]
Thanks, @janiversen 👍

**Pull request merged**